lib/chef/knife/azure_server_create.rb

```diff
@@ -815,6 +815,14 @@ def get_chef_extension_public_params
         pub_config[:client_rb] = "chef_server_url \t #{Chef::Config[:chef_server_url].to_json}\nvalidation_client_name\t#{Chef::Config[:validation_client_name].to_json}"
         pub_config[:runlist] = locate_config_value(:run_list).empty? ? "" : locate_config_value(:run_list).join(",").to_json
         pub_config[:autoUpdateClient] = locate_config_value(:auto_update_client) ? "true" : "false"
+        pub_config[:custom_json_attr] = locate_config_value(:json_attributes) || {}
+
+        # bootstrap attributes
+        pub_config[:client_rb] += "\nenvironment \t #{locate_config_value(:environment).to_json}" if locate_config_value(:environment)
+        pub_config[:client_rb] += "\nvalidation_key  \t #{Chef::Config[:validation_key].to_json}" if Chef::Config[:validation_key]
+        pub_config[:client_rb] += "\nnode_name  \t #{config[:chef_node_name].to_json}" if config[:chef_node_name]
+        pub_config[:client_rb] += "\nencrypted_data_bag_secret  \t #{locate_config_value(:encrypted_data_bag_secret).to_json}" if locate_config_value(:encrypted_data_bag_secret)
+
```

Comment:
I think for now you just pass this options as pub_config[:bootstrap_options] = "value" to extension. 
We are going to use bootstrap context in extension. And bootstrap context will append this options to client_rb depend upon usage.